pkg/authorization/bearer_token.go

```diff
@@ -0,0 +1,75 @@
+package authorization
+
+import (
+	"crypto/rand"
+	"fmt"
+	"os"
+
+	"github.com/golang-jwt/jwt/v5"
+)
+
+const (
+	TokenPathEnvVar = "KUBERNETES_SERVICE_TOKEN_PATH"
+
+	defaultTokenPath = "/var/run/secrets/kubernetes.io/serviceaccount/token"
+)
+
+var inMemoryToken *[]byte
+
+func CreateToken() (string, error) {
+	token := jwt.New(jwt.SigningMethodHS256)
+
+	secretKey, err := getSecretKey()
+	if err != nil {
+		return "", fmt.Errorf("error getting secret key: %v", err)
+	}
+
+	tokenString, err := token.SignedString(secretKey)
+	if err != nil {
+		return "", fmt.Errorf("error signing token: %v", err)
+	}
+
+	return tokenString, nil
+}
+
+func ValidateToken(tokenString string) (bool, error) {
+	token, err := jwt.Parse(tokenString, func(token *jwt.Token) (any, error) {
+		return getSecretKey()
```

Comment:
Could the secret key file changed while the pod is running? if not, then reading from a file is expensive, and I think we should read it on boot, or at least do it only once, and keep the secret in memory (e.g. by using sync.Once).

Reply:
since we were already keeping the generated token in memory, maybe we can also keep this one, and just check if it is populated already

Reply:
yes, exactly.